src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -551,7 +551,19 @@ await using (fs.ConfigureAwait(false))
                 }
             }
 
-            ArchivingUtils.AttemptSetLastWriteTime(destinationFileName, ModificationTime);
+            AttemptSetLastWriteTime(destinationFileName, ModificationTime);
+        }
+
+        private static void AttemptSetLastWriteTime(string destinationFileName, DateTimeOffset lastWriteTime)
+        {
+            try
+            {
+                File.SetLastWriteTime(destinationFileName, lastWriteTime.LocalDateTime); // SetLastWriteTime expects local time
```

Comment:
I'll open an issue to discuss changing this for ZipFile too.

https://github.com/dotnet/runtime/blob/96a0fb1cd6210fc4842f32f549870a1d82e95c6f/src/libraries/Common/src/System/IO/Archiving.Utils.cs#L39

Reply:
As per the discussion in the linked issue, Zip gets away with it because its lastWriteTime always has a zero offset. So no particular reason to change it, although it would be fine (so long as there's a test -- I think there is)